const Faculty = require("../models/faculty.model");exports.getAllFaculties = async (req, res, next) => {    try {        const faculties = await Faculty.find();        res.status(200).json({message: 'All faculties were fetched successfully', faculties});    } catch (error) {        if (!error.statusCode) {            error.status = 500;        }        next(error);    }};exports.getFaculty = async (req, res, next) => {    const id = req.query.id;    try {        const faculty = await Faculty.findById(id).exec();        if (!faculty) {            let error = new Error('Not found!');            error.statusCode = 404;            throw error;        }        res.status(200).json({message: 'Todo item has been successfully fetched', faculty});    } catch (err) {        if (!err.statusCode) {            err.statusCode = 500        }        next(err)    }};exports.deleteFaculty = async (req, res, next) => {    const id = req.query.id;    try {        const faculty = await Faculty.findByIdAndRemove(id);        if (!faculty) {            let error = new Error('Not found!');            error.statusCode = 404;            throw error;        }        console.log('hh')        res.status(200).json({message: 'Todo item has been successfully deleted', faculty});    } catch (err) {        if (!err.statusCode) {            err.statusCode = 500;        }        next(err);    }};exports.addFaculty = async(req, res, next) => {    try {        const {            name,            abbreviation,            description,            linkToWebSite,            image        } = req.body;        const faculty = await Faculty.create({name, abbreviation, description, linkToWebSite, image});        res.status(201).json({message: 'Faculty was successfully created', faculty});    } catch (error) {        if (!error.statusCode) {            error.statusCode = 500;        }        next(error);    }};